### PR TITLE
feature: ARSN-58-add-MD-actionMap

### PR DESCRIPTION
**Pull request by XinLi0207**
1. add  `metadataSearch: 's3:MetadataSearch'` actionMap( action name is different from what defined in the [design](https://github.com/scality/citadel/blob/development/1.0/docs/design/metadata-search.md), which is `mdsearch`. Through observation, I found it's not common to have abbreviation in the action name, so wanted to change to `metadataSearch` to make it more accord with naming convention )
2. add this logic in routeGet.js: if query.search is not null, call metadataSearch
3. bump version

### CODE REVIEW

File: lib/policyEvaluator/utils/actionMaps.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ const sharedActionMap = {
     bypassGovernanceRetention: 's3:BypassGovernanceRetention',
     listMultipartUploads: 's3:ListBucketMultipartUploads',
     listParts: 's3:ListMultipartUploadParts',
+    metadataSearch: 's3:MetadataSearch',
```
Comment (ghost) [inline lib/policyEvaluator/utils/actionMaps.js:33]: In the design we have `s3:mdsearch`, is it expected to use this format now?

**Review by ghost — COMMENTED** [review]
Quick question before approving

File: lib/policyEvaluator/utils/actionMaps.js (hunk 1 above)
Comment (XinLi0207) [reply lib/policyEvaluator/utils/actionMaps.js:33]: wanted to ask your suggestion on this, feel like in action names it's not common to have abbreviation, `s3: MDSearch` seems a little bit weird here
@anurag4DSB  didn't think it through when we decided on this name, can we change the action name defined in the design?

Comment (ghost) [reply lib/policyEvaluator/utils/actionMaps.js:33]: IMO as long as all implementations are compliant with this name and the design is updated in a quick PR, that's ok

**Review by KillianG — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**